go.mod

```diff
@@ -34,7 +34,7 @@ require (
 	github.com/ghodss/yaml v1.0.0
 	github.com/go-kit/kit v0.12.0
 	github.com/go-kit/log v0.2.1
-	github.com/go-sql-driver/mysql v1.6.0
+	github.com/go-sql-driver/mysql v1.7.1
```

Comment:
Upstream nanomdm relies on v1.7.1 of the mysql driver, which added `SQLState` to the `MySQLError` type and modifies the error string [template](https://github.com/go-sql-driver/mysql/pull/1321/files#diff-76a7e0e299c7417bae32d3098e71370980157fe29e68a366671491d147d40572R66). This causes a number of our tests to break. For [example](https://github.com/fleetdm/fleet/actions/runs/5523207172/jobs/10073809666?pr=12721#step:9:13351):
```
20221223174807_AlterHostsTablePendingMDMEnrollments_test.go:74: 
        	Error Trace:	/home/runner/work/fleet/fleet/server/datastore/mysql/migrations/tables/20221223174807_AlterHostsTablePendingMDMEnrollments_test.go:74
        	Error:      	Error "Error 1048 (23000): Column 'osquery_host_id' cannot be null" does not contain "Error 1048: Column 'osquery_host_id' cannot be null"
```